api/v1_projects.inc

```diff
@@ -316,3 +316,60 @@ function api_v1_projects_pagerounds($method, $data, $query_params)
 
     return array_merge(["OCR"], array_keys($Round_for_round_id_));
 }
+
+function api_v1_project_pdflatex($method, $data, $query_params)
+{
+    // make a directory to work in
+    $temp_path = sys_get_temp_dir() . uniqid("/latex_");
+    mkdir($temp_path);
+    if (!is_dir($temp_path)) {
+        throw new UnauthorizedError("Can't create directory");
+    }
+    chdir($temp_path);
+
+    $page_text = array_get($query_params, "text", "no text");
+
+    // we are in our own directory so can use simple names
+    $base_name = "page";
+    $pdf_base_name = "$base_name.pdf";
+    $tex_base_name = "$base_name.tex";
+
+    // incorporate preamble and postamble
+    $preamble =
+'\\documentclass[12pt,oneside]{book}
+\\usepackage{amssymb}
+\\usepackage{amsmath}
+\\begin{document}
+';
+    $postamble = '\\end{document}
+';
+    // TODO incorporate project-specific preamble
+    file_put_contents($tex_base_name, $preamble . $page_text . $postamble);
+
+    $result = [];
+    $retval = -1;
+    $cmd = "pdflatex -no-shell-escape -halt-on-error $tex_base_name";
+    exec($cmd, $result, $retval);
+
+    $enc_pdf = (0 == $retval) ? base64_encode(file_get_contents($pdf_base_name)) : "";
+
+    // delete files no longer needed
+    $files = glob("*");
```

Comment:
We can't enumerate the outputted files ahead to avoid deleting anything by accident?

Reply:
Yes, we could do that. There will always be basename.tex, and, if the compile succeeded, .pdf and .aux. If there was anything else then removing the directory would fail but it would get cleaned up by the system (in 10 days?) since it is in the tmp directory.